kernel/src/drivers/keyboard/mod.rs

```diff
@@ -0,0 +1,200 @@
+//! # Keyboard Driver
+//!
+//! The keyboard driver handles all keyboard related functionality, intended to support both PS/2 and USB.
+//! Currently, only PS/2 support has been implemented through the use of the PS/2 driver.
+//!
+//! The driver is event based, and events are received through the `read_event` method, which blocks until an event is received.
+//! The event contains the keycode pressed, which can be compared to `keymap::codes`, an optional `char`, the type of press, and various modifier flags.
+
```

Comment:
This whole module could benefit from more detailed doc comments and examples. In future we should document and write examples for everything. We will have to go back and write docs for existing features. (Opening issue)

Reply:
Done.